### DISCUSSION TITLE
Need way to trigger switch on IR command

### DISCUSSION THREAD

**Discussion by iz3man** [arendst/Tasmota#17572] [General]
Hi

I had this running before, but have NO CLUE how it was done before.
I got an ESP32 and a switch on D6 with IR-Receiver on D7. D6 goes to a power connector of a PC's mainboard and is when activated starts/stop the PC.
Activation is done by an IR signal it receives on D7. 
I know there was some simply way to solve this, but unfortunately the ESP reset somehow and lost it's config, and I don't have a working backup.

I can see commands coming in through IR:

![image](https://user-images.githubusercontent.com/11553674/210266425-31e0ea19-f761-4f41-921b-6d60fbe283a7.png)

Any one willing to help please?
Thank you

![image](https://user-images.githubusercontent.com/11553674/210266362-701aadfa-52a4-4dae-9e22-8626a3beda0f.png)

**Comment by sfromis** [comment]
Further details depends on the requirements for using what you've connected to on your PC, but you can act on the IR part using rules like this: `on IrReceived#data=0xA8C do Power1 1 endon`
If you want to simulate a short button push, setting `PulseTime` should work well for that.

**Reply by iz3man** [reply]
Perfect. Thank you. Now I remember that it was something like that. Unfortunately it throws an error.

```
11:15:28.992 CMD: set on IrReceived#data=0xA8C do Power1 1 endon
11:15:28.998 RSL: RESULT = {"Command":"Unknown"}
```

But now that I know where to start I should be able to solve it.
THANK YOU very much!!

**Reply by iz3man** [reply]
It needs the RULE command first ;)

**Reply by iz3man** [reply]
Btw: Now that I know it's RULES that I was looking for, I found that there are some present already. So it seems there is something else no working correctly.

```
11:29:18.215 CMD: rule1
11:29:18.222 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":34,"Free":477,"Rules":"on Switch2#state=1 do VAR1 1 endon"}}
11:29:20.840 CMD: rule2
11:29:20.848 RSL: RESULT = {"Rule2":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":99,"Free":412,"Rules":"on IrReceived#Data=0x2278 do if (VAR1==1) power1 OFF endif endon on Switch2#state=0 do VAR1 0 endon"}}
11:29:23.173 CMD: rule3
11:29:23.181 RSL: RESULT = {"Rule3":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":106,"Free":405,"Rules":"on IrReceived#Data=0x2A4C0286B430 do if (VAR1==1) power1 ON endif endon on Switch2#state=0 do VAR1 0 endon"}}
```

**Reply by iz3man** [reply]
FYI: No idea why, but somehow it seems the IR code was not recognized after a firmware upgrade. Changed it to what the IR saw incoming, and it's working again :)
